File: 682-baseball-game/baseball-game.py
class Solution(object):
    def calPoints(self, operations):
        stack=[]
        result=0
        for op in operations:
            if op=="C":
                stack.pop()
            elif op=="D":
                stack.append(stack[-1]*2)
            elif op=="+":
                a=stack[-1]
                stack.pop()
                b=stack[-1]
                stack.append(a)
                stack.append(a+b)                    
            else:
                stack.append(int(op))
        for i in range(len(stack)):
            result+=stack[-1]
            stack.pop()
        return result

File: 682-baseball-game/baseball-game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function(operations) {
    let res=[];
    for(oper of operations){
        if(oper==="D"){
            let newScore=res[res.length-1]*2;
            res.push(newScore);
        }
        else if(oper==="C"){
            res.pop();
        }
        else if(oper==="+"){
            let num1=res[res.length-1];
            let num2=res[res.length-2];
            res.push(num1+num2);
        }
        else{
            res.push(Number(oper));
        }
    }
    let answer=0;
    for(num of res){
        answer+=num;
    }
    return answer;
};